src/mlpack/methods/hdbscan/hdbscan_impl.hpp

```diff
@@ -0,0 +1,755 @@
+ * 2. Now consider all the points inside the obtained BFS
+ *    which have at least two child clusters.
+ *    2.1 Find the left and right child cluster.
+ *    2.2 Find the number of point in both of them.
+ *    2.3 For each child cluster,
+ *        if the size of child cluster  > minPoints
+ *            assign new label to child cluster
+ *            add it as the edge of condensed tree
+ *        else
+ *            assign it and all its children
+ *            in bfs of single linkage tree
+ *            the label of parent cluster,
+ *            set ignore as true for this
+ *            child cluster and all its children in
+ *            the single linkage tree                              
+ *       
+ */
+template<typename NeighborSearch ,
+         typename MetricType ,
+          template<typename TreeMetricType,
+                   typename TreeStatType,
+                   typename TreeMatType> class TreeType >
+template<typename MatType>
+void HDBSCAN<NeighborSearch,  MetricType, TreeType>::condenseTree(const MatType& singleLinkageTree, MatType& result, size_t minClusterSize )
+{
+  std::vector<size_t> bfs;
+  size_t resultCounter = 0;// Keeps track of number of elments in result matrix
+  size_t rootOfBFS = 2 * singleLinkageTree.n_rows;
+  singleLinkageTreeToModifiedBFS(singleLinkageTree, bfs, rootOfBFS);
+  result.set_size(bfs.size() * 2, 4) ;
+  double lambda;
+  size_t noOfPoints = singleLinkageTree.n_rows + 1;
+  size_t nextLabel = noOfPoints + 1;
+  //stores label of all the clusters
+  std::vector<size_t> relabel(2 * singleLinkageTree.n_rows + 1);
+  relabel[rootOfBFS] = noOfPoints;
+  //if cluster i can be aprent or not
+  std::vector<bool> ignore (bfs.size(), false);
+  double delta;
+  size_t leftCount,rightCount,leftChild,rightChild;
+  std::vector<size_t> bfsOfChild;
+
+
+  for(size_t i=0; i<bfs.size(); i++)
+  {  
+      size_t currNode = bfs[i];
+      //currNode cant be a parent
+      if( ignore[currNode] || currNode < noOfPoints) continue;
+     
+      //find left right child clusters, their size and lambda
+      leftChild = singleLinkageTree((currNode - noOfPoints), 0);
+      rightChild = singleLinkageTree((currNode - noOfPoints), 1);
+      leftCount = ( (leftChild >= noOfPoints) ? singleLinkageTree((leftChild - noOfPoints), 3) : 1);
+      rightCount = ( (rightChild >= noOfPoints) ? singleLinkageTree((rightChild - noOfPoints), 3) : 1);
+      delta = singleLinkageTree((currNode - noOfPoints), 2);
+      if(delta > 0.0)
+        lambda = 1.0 / delta;
+      else
+        lambda = std::numeric_limits<double>::infinity();
+
+      if(leftCount >= minClusterSize && rightCount >= minClusterSize)
+      {
+        relabel[leftChild] = nextLabel;
+        nextLabel++;
+        result(resultCounter,0) = relabel[currNode];
+        result(resultCounter,1) = relabel[leftChild];
+        result(resultCounter,2) = lambda;
+        result(resultCounter,3) = leftCount;
```

Comment:
Armadillo is column major, so you should really be updating a column at once, not a row at once.  In either case since you don't have all four values available in an `arma::vec` you probably have to update this manually.